### PR TITLE
Fix/flutter 3.7

### PR DESCRIPTION
**Pull request by StanevPrime**
Set higher OS minimum version and fix static code analysis issues in the example projects
- booking
- counter
- devision
- github_search
- reminders

Delete the news app

Package updates
 - rx_bloc: 4.0.0
 - flutter_rx_bloc: 5.0.2
 - rx_bloc_list: 3.2.1
 - rx_bloc_generator: 6.0.1
 - intellij_plugin

### CODE REVIEW

**Review by PrimeVoxed — COMMENTED** [review]
Please review and address the comments

File: examples/github_search/lib/base/extensions/exception_extensions.dart [hunk 1]
```diff
@@ -21,12 +21,12 @@ String _extractMessageFromDio(DioError error) {
     final errors = error.response?.data['errors'];
 
     if (errors is List) {
-      errors.forEach((resourceError) {
+      for (var resourceError in errors) {
```
[the remark this answers is not in the document]
Comment (StanevPrime) [reply examples/github_search/lib/base/extensions/exception_extensions.dart:24]: Fixed. @PVGeorgievPrime, please check this code snippet in the rx_bloc_cli

File: examples/github_search/lib/base/theme/github_search_theme.dart [hunk 2]
```diff
@@ -43,51 +38,53 @@ class GithubSearchTheme {
...
-      _buildDesignTextTheme(base.accentTextTheme, designSystemColor),
+          _buildDesignTextTheme(base.primaryTextTheme, designSystemColor),
       appBarTheme: AppBarTheme(
         color: designSystemColor.primaryVariant,
       ),
       iconTheme: _buildIconTheme(base.iconTheme, designSystemColor),
+      colorScheme: colorScheme
+          .copyWith(error: designSystemColor.errorColor)
+          .copyWith(background: designSystemColor.backgroundColor)
+          .copyWith(secondary: designSystemColor.accentColor),
```
[the remark this answers is not in the document]
Comment (StanevPrime) [reply examples/github_search/lib/base/theme/github_search_theme.dart:49]: Done

File: examples/github_search/lib/feature_login/ui_components/login_form.dart [hunk 3]
```diff
@@ -40,7 +40,9 @@ class _LoginFormState extends State<LoginForm> {
   @override
   void dispose() {
     // Since the focus nodes are long living object, they should be disposed.
-    _focusNodes.forEach((focusNode) => focusNode.dispose());
+    for (var focusNode in _focusNodes) {
```
[the remark this answers is not in the document]
Comment (StanevPrime) [reply examples/github_search/lib/feature_login/ui_components/login_form.dart:43]: Done

File: packages/flutter_rx_bloc/example/android/app/build.gradle [hunk 4]
```diff
@@ -26,7 +26,7 @@ apply plugin: 'kotlin-android'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 android {
-    compileSdkVersion 29
+    compileSdkVersion 31
```
[the remark this answers is not in the document]
Comment (StanevPrime) [reply packages/flutter_rx_bloc/example/android/app/build.gradle:29]: Done

File: packages/rx_bloc_list/example/android/app/build.gradle [hunk 5]
```diff
@@ -25,7 +25,7 @@ apply plugin: 'com.android.application'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 android {
-    compileSdkVersion 30
+    compileSdkVersion 31
```
[the remark this answers is not in the document]
Comment (StanevPrime) [reply packages/rx_bloc_list/example/android/app/build.gradle:28]: Done

**Review by PrimeVoxed — APPROVED** [review]

**Pull request merged**